### PR TITLE
fix: No-test runs now explain likely test assembly setup issues

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- `uloop run-tests` now gives actionable asmdef hints when an unfiltered run discovers zero tests.
- Narrow filter misses still keep the plain no-tests message, so normal exact, regex, and assembly misses are not over-diagnosed.

## User Impact
- Before this change, Unity Test Runner could report `No tests found matching the specified filter criteria` with no clue that the project asmdef setup was likely wrong.
- After this change, agents and developers get targeted hints for likely missing test assembly markers, duplicate TestRunner references, wrong EditMode or PlayMode platform targeting, and missing runtime asmdef references.

## Changes
- Adds no-test diagnostics that run only for unfiltered zero-discovery results.
- Detects common test asmdef layouts, including nested feature test assemblies.
- Keeps the run-tests response schema unchanged and updates generated run-tests skill docs.
- Adds focused EditMode coverage for the diagnostic and false-positive guard cases.

## Verification
- `git diff --check`
- `dotnet build UnityCLILoop.FirstPartyTools.RunTests.Editor.csproj /v:minimal /p:RunAnalyzers=false`
- `cli/dist/darwin-arm64/uloop compile --project-path <PROJECT_ROOT>`
- `cli/dist/darwin-arm64/uloop run-tests --project-path <PROJECT_ROOT> --test-mode EditMode --filter-type regex --filter-value RunTestsNoTestsDiagnosticServiceTests`
- `cli/dist/darwin-arm64/uloop run-tests --project-path <PROJECT_ROOT> --test-mode EditMode --filter-type exact --filter-value Does.Not.Exist`
- `codex-review v3-beta`

### CODE REVIEW

File: Packages/src/Editor/FirstPartyTools/RunTests/RunTestsUseCase.cs [hunk 1]
```diff
@@ -117,6 +119,13 @@ public async Task<UnityCliLoopTestExecutionResult> ExecuteAsync(UnityCliLoopTest
                 SkippedCount = result.skippedCount,
                 XmlPath = result.xmlPath
             };
+            response.Message = _noTestsDiagnosticService.AppendDiagnosticsIfNeeded(
+                response.Message,
+                response.Success,
+                response.TestCount,
+                parameters.TestMode,
+                parameters.FilterType);
```
[the remark this answers is not in the document]
Comment (qua-hatayama) [reply Packages/src/Editor/FirstPartyTools/RunTests/RunTestsUseCase.cs]: Fixed in b351e138.

The comment was valid: no-test asmdef diagnostics are optional and should not replace the original run-tests result if diagnostic construction fails.

I added a best-effort wrapper that logs diagnostic failures and returns the original message, then routed the use case through it. I also added focused coverage for the failure path.

Verification:
- git diff --check
- dotnet build UnityCLILoop.FirstPartyTools.RunTests.Editor.csproj /v:minimal /p:RunAnalyzers=false
- cli/dist/darwin-arm64/uloop compile --project-path <PROJECT_ROOT>
- cli/dist/darwin-arm64/uloop run-tests --project-path <PROJECT_ROOT> --test-mode EditMode --filter-type regex --filter-value RunTestsNoTestsDiagnosticServiceTests
- cli/dist/darwin-arm64/uloop run-tests --project-path <PROJECT_ROOT> --test-mode EditMode --filter-type exact --filter-value Does.Not.Exist

File: Packages/src/Editor/FirstPartyTools/RunTests/RunTestsNoTestsDiagnosticService.cs [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+
+            try
+            {
+                return appendDiagnostics();
+            }
+            catch (Exception exception)
+            {
+                Debug.LogWarning($"Failed to build no-tests diagnostics: {exception}");
+                return message;
+            }
```
[the remark this answers is not in the document]
Comment (qua-hatayama) [reply Packages/src/Editor/FirstPartyTools/RunTests/RunTestsNoTestsDiagnosticService.cs:54]: Fixed in 4ef46e36.

I narrowed the no-test diagnostics fallback so it only downgrades recoverable asmdef inspection failures (I/O and access failures). Logic failures such as InvalidOperationException now propagate instead of being hidden behind the original no-tests message, and the focused EditMode tests cover both paths.
